tests/test_batchupload.py

```diff
@@ -35,6 +34,41 @@ def test_cancel(server):
         batch.get(0)
 
 
+@pytest.mark.parametrize('hash, is_valid', [
+    # Known algos
+    ('0' * 32, False),
+    ('0' * 40, False),
+    ('0' * 56, False),
+    ('0' * 64, False),
+    ('0' * 96, False),
+    ('0' * 128, False),
+    # Other
+    (None, True),
+    ('', True),
+    ('foo', True),
+])
+def test_digester(hash, is_valid, server):
+    file_out = 'test_out'
+    doc = server.documents.create(new_doc, parent_path=pytest.ws_root_path)
+    try:
+        batch = get_batch(server)
+        operation = server.operations.new('Blob.AttachOnDocument')
+        operation.params = {'document': pytest.ws_root_path + '/Document'}
+        operation.input_obj = batch.get(0)
+        operation.execute(void_op=True)
+
+        operation = server.operations.new('Blob.Get')
+        operation.input_obj = pytest.ws_root_path + '/Document'
+        if is_valid:
+            operation.execute(file_out=file_out, digest=hash)
+        else:
+            with pytest.raises(CorruptedFile):
+                operation.execute(file_out=file_out, digest=hash)
+    finally:
+        os.remove(file_out)
+        doc.delete()
```

Comment:
Could you reverse those 2 lines? It is preferable to delete the remote document first in case the local delete fails.